File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        int[] arr = new int['Z'-'A'+1];
        Arrays.fill(arr, 200);
        for(int i=0; i<keymap.length; i++){
            for(int j=0; j<keymap[i].length(); j++){
                int index = keymap[i].charAt(j) - 'A';
                arr[index] = Math.min(arr[index], j+1);
            }
        }
        for(int i=0; i<targets.length; i++){
            for(int j=0; j<targets[i].length(); j++){
                int index = targets[i].charAt(j) - 'A';
                if(arr[index] <= 100){
                    answer[i] += arr[index];
                } else{
                    answer[i] = -1;
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = [0] * len(targets)
    arr = [101] * 26
    for i in range(len(keymap)):
        for j in range(len(keymap[i])):
            index = ord(keymap[i][j]) - ord('A')
            arr[index] = min(arr[index], j+1)
    for i in range(len(targets)):
        for j in range(len(targets[i])):
            index = ord(targets[i][j]) - ord('A')
            if arr[index] > 100:
                answer[i] = -1
                break
            answer[i] += arr[index]
    return answer